search/query/knn.go

```diff
@@ -35,7 +35,10 @@ type KNNQuery struct {
 	BoostVal    *Boost    `json:"boost,omitempty"`
 
 	// see KNNRequest.Params for description
-	Params json.RawMessage `json:"params"`
+	Params           json.RawMessage `json:"params"`
+	FilterQuery      Query           `json:"filter,omitempty"`
+	FilterResults    []index.IndexInternalID
```

Comment:
Does this need exporting?

Reply:
I've changed the approach to not export these now.